examples/neural_compressor/optical-character-recognition/run_trocr.py

```diff
@@ -0,0 +1,310 @@
+            raise ValueError("Need a dataset path")
+
+
+@dataclass
+class OptimizationArguments:
+    """
+    Arguments pertaining to what type of optimization we are going to apply on the model.
+    """
+
+    apply_quantization: bool = field(
+        default=False,
+        metadata={"help": "Whether or not to apply quantization."},
+    )
+    quantization_approach: Optional[str] = field(
+        default=None,
+        metadata={"help": "Quantization approach. Supported approach are static, dynamic and aware_training."},
+    )
+    quantization_config: Optional[str] = field(
+        default=None,
+        metadata={
+            "help": "Path to the directory containing the YAML configuration file used to control the quantization and "
+            "tuning behavior."
+        },
+    )
+    tolerance_criterion: Optional[float] = field(
+        default=None,
+        metadata={"help": "Performance tolerance when optimizing the model."},
+    )
+    verify_loading: bool = field(
+        default=False,
+        metadata={"help": "Whether or not to verify the loading of the quantized model."},
+    )
+    warmup_iter: int = field(
+        default=5,
+        metadata={"help": "For benchmark measurement only."},
+    )
+
+
+def main():
+    parser = HfArgumentParser((ModelArguments, DataTrainingArguments, TrainingArguments, OptimizationArguments))
+    if len(sys.argv) == 2 and sys.argv[1].endswith(".json"):
+        # If we pass only one argument to the script and it's the path to a json file,
+        # let's parse it to get our arguments.
+        model_args, data_args, training_args, optim_args = parser.parse_json_file(
+            json_file=os.path.abspath(sys.argv[1])
+        )
+    else:
+        model_args, data_args, training_args, optim_args = parser.parse_args_into_dataclasses()
+
+    # Setup logging
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        handlers=[logging.StreamHandler(sys.stdout)],
+    )
+
+    log_level = training_args.get_process_log_level()
+    logger.setLevel(log_level)
+    transformers.utils.logging.set_verbosity(log_level)
+    transformers.utils.logging.enable_default_handler()
+    transformers.utils.logging.enable_explicit_format()
+
+    # Log on each process the small summary:
+    logger.warning(
+        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
+        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
+    )
+    logger.info(f"Training/evaluation parameters {training_args}")
+
+    test_file = os.path.join(data_args.datasets_dir, "gt_test.txt")
+    test_dir = os.path.join(data_args.datasets_dir, "image")
+    df = pd.read_fwf(test_file, header=None)
+    df.rename(columns={0: "file_name", 1: "text"}, inplace=True)
+    del df[2]
+    df.head()
+
```

Comment:
We should add `data_args.max_calibration_samples` and `data_args.max_eval_samples` to control the number of calibration and evaluation samples

Reply:
Done